daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf_1.proto

```diff
@@ -1337,6 +1337,28 @@ message Update {
     Expr arg = 5;
   }
 
+  // Soft Exercise Update
+  // Exercises a contract of the given template type OR a predecessor.
+  // *Available in versions >= 1.dev*
+  message SoftExercise {
+    // Template type
+    TypeConName template = 1;
+
+    // name of the exercised template choice
+    oneof choice {
+      // *Must be a valid identifier*
+      string choice_str = 2;
+
+      // *Must be a valid interned identifier*
+      int32 choice_interned_str = 6;
+    }
+
+    // contract id
+    Expr cid = 3;
+    // argument
+    Expr arg = 5;
+  }
```

Comment:
Since this feature is newer than string interning, I think we can simplify this a bit (this will need a similar change in the encoders/decoders)

```suggestion
  message SoftExercise {
    // Template type
    TypeConName template = 1;

    // name of the exercised template choice
    // *Must be a valid interned identifier*
    int32 choice_interned_str = 2;

    // contract id
    Expr cid = 3;
    // argument
    Expr arg = 4;
  }
```

Reply:
I'm not sure this simplification is a good idea. because of the changes it causes in the encoders/decoders.

Reply:
keeping everything is the same is simpler.
Maybe one day we can kill all the non-interning cases in one go.